src/js/libgeo/GeoOps.js

```diff
@@ -625,6 +625,144 @@ geoOps.PointOnSegment.updatePosition = function(el) {
 };
 geoOps.PointOnSegment.stateSize = 2;
 
+geoOps._helper.PointOnArcCr = function(arc,P){
+    var A = arc.startPoint, B = arc.viaPoint, C = arc.endPoint;
+    //var cr = List.crossratio3harm(P,A,B,C,List.ii);
+    var cr = List.crossratio3harm(A,C,B,P,List.ii);
+    var m = cr.value[0];
+    var n = cr.value[1];
+    if (!CSNumber._helper.isAlmostZero(m)) {
+        n = CSNumber.div(n, m);
+        m = CSNumber.real(1);
+    } else {
+        m = CSNumber.div(m, n);
+        n = CSNumber.real(1);
+    }
+    return [m,n];
+};
+
+geoOps._helper.projectPointToArc = function(arc,P){
+    var conic = arc.matrix;
+    var cen = geoOps._helper.CenterOfConic(conic);
+
+    var diam = List.cross(cen, P);
+ //   Render2D.drawline(diam);
+//    debugger;
+
+    var isec = geoOps._helper.IntersectLC(diam, arc.matrix);
+
+
+    var crA = geoOps._helper.PointOnArcCr(arc, isec[0]);
+    var prodA = CSNumber.mult(crA[0], crA[1]);
+   // console.log("pA",niceprint(prodA));
+    if (crA[0].value.real < 0) prodA = CSNumber.neg(prodA);
```

Comment:
It is slightly confusing to have `A` and `B` denote the two points of intersection here, since in al the other code `A`, `B` and `C` form the projective basis. Perhaps name these points here `P` and `Q`?